core/crypto/signed_hdr.c

```diff
@@ -52,7 +52,63 @@ struct shdr *shdr_alloc_and_copy(size_t offs, const void *img, size_t img_size)
 static bool is_weak_hash_algo(uint32_t algo)
 {
 	return algo == TEE_ALG_MD5 || algo == TEE_ALG_SHA1 ||
-	       algo == TEE_ALG_MD5SHA1;
+	       algo == TEE_ALG_MD5SHA1 || algo == TEE_ALG_DSA_SHA224 ||
+	       algo == TEE_ALG_HMAC_SHA224 ||
+	       algo == TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA224 ||
+	       algo == TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224 ||
+	       algo == TEE_ALG_RSASSA_PKCS1_V1_5_SHA224 ||
```

Comment:
These are not `hash_algo`, they are `shdr->algo`. So, this compiles, but isn't correct, since I believe that this call in `shdr_verify_signature`
```c
	hash_algo = TEE_DIGEST_HASH_TO_ALGO(shdr->algo);
```
will never return these three. I added it still to get comments from the rest of you. If you have any ideas how to approach this.

Reply:
It should be enough to add `algo == TEE_ALG_SHA224`, all the combined algorithms should be detected automatically.